src/tensorflow.cc

```diff
@@ -1286,21 +1287,29 @@ AutoCompleteHelper::FixBatchingSupport()
     }
   }
 
-  // Set max-batch-size to 1 if the model signature and config hint
+  // max_batch_size = 0 is ambiguous as to whether or not the
+  // user set this value in the config or omitted it. Assume
+  // omitted if model supports batching and check
+  // default_max_batch_size.
+  if (model_support_batching_ && max_batch_size == 0) {
+    max_batch_size =
+        std::max(model_state_->BackendConfig()->default_max_batch_size_, 0);
+  }
+
+  // Set max-batch-size to current value if the model signature and config hint
   // agree. We need to update the configuration itself as well as the
   // cached value we have already initialized in the model state.
-  if (max_batch_size == 0) {
-    const int new_max_batch_size = model_support_batching_ ? 1 : 0;
-
+  if (max_batch_size != 0) {
```

Comment:
Currently, this logic will run even if user has already provided a max_batch_size in their config.pbtxt.
Why not replace L1289-L1303 to use the previous construct itself? Something like below:

```
if (max_batch_size == 0) {
    const int new_max_batch_size = model_support_batching_ ? std::max(model_state_->BackendConfig()->default_max_batch_size_, 0) : 0;
..
..
..
```

Reply:
After looking at this again, I thought I had a good reason for splitting this out, but I think I had a wrong idea. Reverted back to the ternary operator implementation.